### ISSUE TITLE
Ledger Bitcoin App 2.1.1 sign psbt does not contribute a signature

### ISSUE THREAD

**Issue by GambolingPangolin** [bitcoin-core/HWI#671]
Using HWI 2.2.1 with Ledger Bitcoin App 2.1.1 to sign a P2SH-P2WSH multisig output does not work.  There is no error, but the PSBT comes back from the call without the expected signature.  Furthermore, on the device the signing workflow is not triggered.

**Comment by GambolingPangolin** [comment]
Yes, the PSBT does have both fields set.  It is not privacy sensitive, so I have attached the file here.

[example.psbt.txt](https://github.com/bitcoin-core/HWI/files/11080240/example.psbt.txt)

**Comment by bigspider** [comment]
The psbt is missing the global xpubs, which the app relies on to identify the internal keys.
Hopefully adding those will allow signing successfully.

**Comment by bigspider** [comment]
I'm referring to the `PSBT_GLOBAL_XPUB` fields in the global map, per [BIP-174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki); HWI uses them to reconstruct the exact multisig policy, and the app can't sign without that knowledge, because the multisig has to registered on the device first.

(It might have worked with versions of the Ledger bitcoin app before 2.1.0, but that wasn't very secure; from version 2.1.0, the multisig policy registration is compulsory − but HWI should be able to handle it)

**Comment by GambolingPangolin** [comment]
Thanks for the advice!  I'll give that a try.

**Comment by Sjors** [comment]
I'm encountering something similar with a (native segwit) P2WSH multisig. Using the latest HWI master and the Ledger Bitcoin App 2.1.1. It seems that Bitcoin Core is not setting the `global_xpubs` field when you call `walletprocesspsbt`. cc @achow101

**Comment by Sjors** [comment]
A workaround is to use Specter Desktop.

**Comment by KayBeSee** [comment]
> because the multisig has to registered on the device first.

How does one go about registering the multisig on the device? I'm not seeing the register command in the docs. Can this be done via HWI?

cc @bigspider

**Comment by bigspider** [comment]
> > because the multisig has to registered on the device first.
> 
> How does one go about registering the multisig on the device? I'm not seeing the register command in the docs. Can this be done via HWI?
> 
> cc @bigspider

HWI doesn't currently have a specific API for that, as registration returns an hmac that is the responsibility of the caller to store. Therefore, at this time it registers the policy every time you show an address / spend from a multisig − which is redundant but at least it works.

I made a [PoC of full integration](https://github.com/bitcoin-core/HWI/pull/647) some time back that also includes miniscript support.

**Comment by Sjors** [comment]
See #792.
